### PR TITLE
Kaoru freds device id apis

### PR DESCRIPTION
**Pull request by KaoruDev**
ok, so it turns out `1.8` was never supported, as to why travis ran a `1.8` spec, no idea. Changes I've made:
- reorganize classes / files a bit
- add specs for updating device labels
- have travis run specs for rubies between `1.9` - `2.3`
- implement `Error` hierarchy
- wrap `REST` interaction inside of one class `RestWrapper`.

### CODE REVIEW

**Comment by KaoruDev** [comment]
@mwylde would love your thoughts on this

**Comment by KaoruDev** [comment]
@fredsadaghiani can I haz review plz?

File: spec/helpers/mock_responses/devise.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
```
Comment (fredsadaghiani) [inline spec/helpers/mock_responses/devise.json]: device.json?

**Comment by fredsadaghiani** [comment]
Can you please update README.md with examples of the new APIs? Also, the README.md seems out of date.

**Pull request closed without merging**